freqtrade/freqtradebot.py

```diff
@@ -104,6 +105,16 @@ def __init__(self, config: Dict[str, Any]) -> None:
         self._exit_lock = Lock()
         LoggingMixin.__init__(self, logger, timeframe_to_seconds(self.strategy.timeframe))
 
+        if 'trading_mode' in self.config:
+            self.trading_mode = self.config['trading_mode']
+        else:
+            self.trading_mode = TradingMode.SPOT
+
+        if self.trading_mode == TradingMode.FUTURES:
+            for time_slot in self.exchange.funding_fee_times:
+                schedule.every().day.at(time_slot).do(self.update_funding_fees())
+                self.wallets.update()
```

Comment:
I think we will want a test specifically covering this (that the `update_funding_fees()` method is called when defined).

Please look at `time_machine` for this (it's in some other tests ...) - it provides a very nice and simple API to "move" time around ... so moving time "past" the expected point and running `run_pending` should show it's called (ideally, we can also check if the calculation is correct for 1 or more "missing" periods using this method).

Reply:
I tried a little bit

Reply:
> we can also check if the calculation is correct for 1 or more "missing" periods using this method).

Not until the backtesting part is implemented

Reply:
@rokups Do you know the right way to write this test?

Reply:
i've fixed the "nirvana-mock" now - however the test is still failing as the method is never called.

I suspect you'll have to call `schedule.run_pending()` someplace (either directly in the test, or better (most likely) in `process()`.

at the moment, it's only called in `startup_update_open_orders()`- which is only called during startup, but never again while the bot is running.

Reply:
I tried again, they're still getting called 0 times

Reply:
It's a pretty simple test ... simple debugging will show you that:
* you create the schedule "today" (october)
* you THEN move the time to september - and expect the job to be triggered (which is not going to happen).

Simply moving the "time_machine" stuff to the top will help with that (i've however also changed the way we use time_machine - as in this case, i like the fixture "magic" better).

Reply:
I got the tests we had working, but added 3 more that are failing for some reason.